### PR TITLE
Parse reek output line by line

### PR DESCRIPTION
**Pull request by bloomrain**
This fix should solve #11 issue. Before we put put all reek output in one object that was passed to linter. Now we split reek output by lines ar parse every line separately

### CODE REVIEW

File: lib/linter-reek.coffee [hunk 1]
```diff
@@ -30,23 +31,16 @@ module.exports =
...
           filePath = TextEditor.getPath()
-          data = []
+          parsedOutput = null
           process = new BufferedProcess
             command: @executablePath
             args: [filePath]
             stdout: (output) ->
               console.log output
-              data.push output
+              parsedOutput = new OutputParser(output, filePath)
```
Comment (steelbrain) [inline lib/linter-reek.coffee]: How about you use the `atom-linter` npm module's exec method?

File: lib/linter-reek.coffee [hunk 2]
```diff
@@ -28,29 +30,8 @@ module.exports =
...
-
-          process.onWillThrowError ({error,handle}) ->
-            atom.notifications.addError "Failed to run #{@executablePath}",
-            detail: "#{error.message}"
-            dismissable: true
-            handle()
-            resolve
+        filePath = TextEditor.getPath()
+        helpers.exec(@executablePath, [filePath]).then (output)->
+          console.log output
```
Comment (steelbrain) [inline lib/linter-reek.coffee]: can we please drop this `console.log` here?

**Comment by steelbrain** [comment]
Is creating a separate class for output parser really necessary? We also have a `parse` method in the helper module that parses it line-by-line via a provided regex.

**Comment by bloomrain** [comment]
It looked more OOP until you told me about atom-linter parse method

**Comment by steelbrain** [comment]
Thanks! Now all we need is someone testing this and confirming that it works.

**Comment by dhagerty9009** [comment]
Merged in.

**Pull request closed without merging**